dag_in_context/src/optimizations/switch_rewrites.egg

```diff
@@ -29,6 +31,34 @@
       ((union (Get if_e k) (Bop (Smax) a b)))
       :ruleset switch_rewrite) 
 
+; if pred then a else b ~~> (select pred a b)
+; where a and b are inputs to the region
+(rule (
```

Comment:
I think it's possible for this rule to fork memory if memory is passed through in both branches
To fix this, just make sure that the indices for the arg are not equal